### PR TITLE
Update rest_client.rb to support 'rest' parameter

### PR DESCRIPTION
**Pull request by ldep30**
Pull request to fix [https://github.com/ytti/oxidized/issues/466](https://github.com/ytti/oxidized/issues/466)

### CODE REVIEW

**Comment by ytti** [comment]
You're creating new config, shouldn't you load the existing config and use what ever url there is in config?

**Comment by ldep30** [comment]
Is there an existing Ruby "Oxidized config" object I could use ?

I've tried to use something like :

`restcfg = Oxidized.config.rest`

but it leads to an error : 

```in `<class:RestClient>': undefined method `config' for Oxidized:Module (NoMethodError)```

Could you point me to the right direction ? Thanks

**Comment by ytti** [comment]
I believe ```CFG``` should be usable. I.e. ```CFG.rest```.

https://github.com/ytti/oxidized/edit/master/extra/syslog.rb#L46

**Comment by ldep30** [comment]
I had already tried that, an error is generated too :

```in `<class:RestClient>': uninitialized constant Oxidized::RestClient::CFGS (NameError)```

This error is triggered when the config code is just reduced to 
```
CFG = CFGS.cfg
restcfg = CFG.rest
```
Same error with just 
```
restcfg = CFG.rest
```

error : ```in `<class:RestClient>': uninitialized constant Oxidized::RestClient::CFG (NameError)```

I don't know Ruby much but I think the object CFGS is not passed to RestClient when called from Syslog.

**Comment by ytti** [comment]
Capitals are const/global, so if it had been loaded before, it should be visible.

So ```CFG.rest```or ```::Oxidized::CFG.rest``` should be visible, provided ```syslog.rb``` is ran first, before it is referred to in rest_client. 
If ```CFG``` is referred in rest_client.rb definition (loaded during ```require ´rest_client'````) then it won't work, as it's not defined yet.

So use the ```CFG``` inside methods and it should work.

Or, maybe even better, just redefine it, like in ```syslog.rb``` so that ```rest_client.rb``` can work correctly, when when ```syslog.rb``` has not been loaded.

File: extra/rest_client.rb [hunk 1]
```diff
@@ -2,8 +2,32 @@ module Oxidized
   class RestClient
     require 'net/http'
     require 'json'
-    HOST = 'localhost'
-    PORT = 8888
+    require 'uri'
+    require 'asetus'
+
+    class Config
+      Root      = File.join ENV['HOME'], '.config', 'oxidized'
```
Comment (danilopopeye) [inline extra/rest_client.rb]:
``` ruby
Root = ENV['OXIDIZED_HOME'] || File.join(ENV['HOME'], '.config', 'oxidized')
```

File: extra/rest_client.rb [hunk 2]
```diff
@@ -2,8 +2,32 @@ module Oxidized
...
+    begin
+      CFGS.load
+    rescue => error
+      raise InvalidConfig, "Error loading config: #{error.message}"
+    ensure
+      CFG = CFGS.cfg
+    end
+
+    restcfg = CFG.rest
+    if ! restcfg.match(/^http:\/\//)
```
Comment (danilopopeye) [inline extra/rest_client.rb]: `unless` isn't better?

File: extra/rest_client.rb [hunk 3]
```diff
@@ -2,8 +2,32 @@ module Oxidized
...
+    end
+
+    CFGS = Asetus.new :name=>'oxidized', :load=>false, :key_to_s=>true
+    CFGS.default.rest = '127.0.0.1:8888'
+
+    begin
+      CFGS.load
+    rescue => error
+      raise InvalidConfig, "Error loading config: #{error.message}"
+    ensure
```
Comment (danilopopeye) [inline extra/rest_client.rb]: I don't get why we need the `ensure` here, @ldep30. care to explain pls?

**Comment by ldep30** [comment]
Hi danilopopeye,

Thanks for the tips. I've added another commit with your 2 suggestions above.

About the "ensure" statement : basically, this bit of code is just a copy/paste of extra/syslog.rb to load the config. I'm a total Ruby beginner so code refactoring is probably necessary :)

I've also tried ytti suggestions about not reloading the whole config and using the CFG object but that was a fail.

Please feel free to submit your modifications here and I'll edit the code.

File: extra/rest_client.rb (hunk 3 above)
Comment (danilopopeye) [inline extra/rest_client.rb]: following [your comment](https://github.com/ytti/oxidized/pull/467#issuecomment-226261488), you don't need this `ensure`.

``` ruby
    begin
      CFGS.load
    rescue => error
      raise InvalidConfig, "Error loading config: #{error.message}"
    end

    restcfg = CFGS.cfg.rest
```

Comment (ldep30) [inline extra/rest_client.rb]: Thanks. Changes are committed.

**Pull request merged**